sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -373,6 +373,33 @@ public static String stringJoin(String delimiter, List<String> values) {
         }
     }
 
+    /**
+     * Create a parameterized type from a raw class and its type arguments.
+     *
+     * @param rawClass The raw class to construct the parameterized type.
+     * @param genericTypes The generic arguments.
+     *
+     * @return The parameterized type.
+     */
+    public static ParameterizedType createParameterizedType(Type rawClass, Type... genericTypes) {
```

Comment:
This already exists in the file but is named `getParameterizedType`, lets rename the other.

And a tangential thought, should we have an implementation class somewhere that is something like `ParameterizedTypeImpl implements ParameterizedType`. If this is called a lot this will generate a lot of anonymous classes (which I don't know if that's fine or causes any odd side effects in the JVM).

Reply:
Do you want to open an issue for the follow-up thought?

Reply:
Sure, will open an issue